PaintTests/PaintTests.cpp

```diff
@@ -12,26 +14,65 @@ using namespace paint;
 namespace PaintTests
 {
 	
-	TEST_CLASS(TestPaintTestElement)
+	//TEST_CLASS(TestPaintTestElement)
+	//{
+	//public:
+
+	//	TEST_METHOD(TestDrawWindowColourSwap)
+	//	{
+	//		/*const auto testWindow = SDL_CreateWindow("rootWindow", 300, 100, 1200, 800, 0);
+	//		const auto renderer = SDL_CreateRenderer(testWindow, -1, SDL_RENDERER_ACCELERATED);
+	//		const auto surface = SDL_GetWindowSurface(testWindow);
+
+	//		gfx::Rectangle drawRect(200, 40, 1000, 720);
+	//		auto drawWindow = std::make_shared<DrawWindow>(testWindow, renderer, surface, drawRect, "drawWindow");
+	//		//auto drawWindow = std::make_shared<DrawWindow>(testWindow, renderer, surface, drawRect, "drawWindow");
+	//		//const gfx::Colour drawColour{ 255, 255, 255, 255 };
+	//		//drawWindow->setBackgroundColour(drawColour);
+
+
+	//		SDL_DestroyWindow(testWindow);
+	//		*/
+	//	}
+	//};
+
+	TEST_CLASS(TestPaintBrush)
 	{
 	public:
 
-		TEST_METHOD(TestDrawWindowColourSwap)
+		TEST_METHOD(TestConstructionInvariant)
+		{
+			const Brush brush(7);
+			//Assert::AreEqual(brush.getThickness(), 7);
+		}
+
+		TEST_METHOD(TestCopyConstruction)
+		{
+			const Brush a(7);
+			const Brush b(a);
+			Assert::AreEqual(b.getThickness(), 7);
+		}
+
+		TEST_METHOD(TestAssignmentOperator)
+		{
+			const Brush a(7);
+			const Brush b = a;
+			Assert::AreEqual(b.getThickness(), 7);
+		}
+
+		TEST_METHOD(TestSetThickness)
 		{
-			/*const auto testWindow = SDL_CreateWindow("rootWindow", 300, 100, 1200, 800, 0);
-			const auto renderer = SDL_CreateRenderer(testWindow, -1, SDL_RENDERER_ACCELERATED);
-			const auto surface = SDL_GetWindowSurface(testWindow);
-			
-			gfx::Rectangle drawRect(200, 40, 1000, 720);
-			auto drawWindow = std::make_shared<DrawWindow>(testWindow, renderer, surface, drawRect, "drawWindow");
-			//auto drawWindow = std::make_shared<DrawWindow>(testWindow, renderer, surface, drawRect, "drawWindow");
-			//const gfx::Colour drawColour{ 255, 255, 255, 255 };
-			//drawWindow->setBackgroundColour(drawColour);
-			
-
-			SDL_DestroyWindow(testWindow);
-			*/
+			Brush a(7);
+			a.setThickness(5);
+			Assert::AreEqual(a.getThickness(), 7);
```

Comment:
Shouldn't this be 5?

Reply:
Fixed and exceptions added to catch thicknesses set to values other than 0, 1, 2. More tests added.